pkg/provisioner/provisioner.go

```diff
@@ -20,7 +20,7 @@ type Factory func(host *metal3v1alpha1.BareMetalHost, bmcCreds bmc.Credentials,
 
 // UserDataSource is the interface for a function to retrieve user
 // data for a host being provisioned.
-type UserDataSource func() (string, error)
+type UserDataSource func() (string, map[string]interface{}, error)
```

Comment:
Combining the user data and network data in 1 secret isn't going to work in OpenShift. The user data secrets there are managed by a part of the system completely separate from metal3. So, I think we'll need to add a new field to the Host CRD to allow a separate network data secret. It will still be possible for both the user data secret and network data secret to refer to the same secret CR, but in cases where they can't be the same we will be able to use different CRs.

Doing it that way will make it more obvious that we need a new method in this interface instead of changing UserDataSource. The new NetworkDataSource will look a lot like UserDataSource but return a map instead of a string.